### PR TITLE
Wdr vent tickets

### PR DESCRIPTION
**Pull request by amihaila**
Submitting a PR since you should look at any new code since the last WDR tag.
This addresses all of my vent code-related basecamp tickets for WDR.

Things I specifically want checked:
- Are the battery limits I set acceptable?
- Is the linear actuator open/close potentiometer tolerance ok?
- Are the linear actuator pot limits correct?
+ Anything else you notice.

### CODE REVIEW

File: error_checks.c [hunk 1]
```diff
@@ -72,14 +90,14 @@ bool check_bus_current_error(void){
 
 bool check_valve_pin_error(enum VALVE_STATE req_state) {
     return true;
-    
+
```
Comment (akmorrison) [inline error_checks.c:93]: Have I got an astylerc for you

File: error_checks.c [hunk 2]
```diff
@@ -43,13 +45,29 @@ bool check_battery_voltage_error(void){    //returns mV
         can_msg_t error_msg;
         build_board_stat_msg(timestamp, error_code, batt_data, 2, &error_msg);
         txb_enqueue(&error_msg);
+
+        // main loop should check this and open the vent valve if needed
+        if (batt_voltage_mV < VENT_BATT_UNDERVOLTAGE_PANIC_THRESHOLD_mV) {
```
Comment (akmorrison) [inline error_checks.c:50]: I feel like there should be some amount of hysteresis on this, something like "we need to read 10 good battery voltages in a row before we consider the voltage to no longer be critical". I worry that the battery will get to exactly the threshold voltage, and then noise on the ADC will cause us to toggle the vent every quarter second.

There's no major downside to toggling it like that (bigger current draw, maybe stressing the MOSFETs a bit), but still.

File: lin_actuator.c [hunk 3]
```diff
@@ -21,10 +21,6 @@ bool lin_actuator_dac_init(void) {
   data_buf[1] = ((HIGH_DAC_VAL & 0x0f) << 4);
   i2c1_writeNBytes(HIGH_DAC_ADDRESS, data_buf, 2);
 
-  data_buf[0] = ((LOW_DAC_VAL & 0xf0) >> 4);
-  data_buf[1] = ((LOW_DAC_VAL & 0x0f) << 4);
-  i2c1_writeNBytes(LOW_DAC_ADDRESS, data_buf, 2);
-
```
Comment (akmorrison) [inline lin_actuator.c]: I would have sworn that I committed this yesterday. Wanna rebase onto origin/master?

File: lin_actuator.c [hunk 4]
```diff
@@ -69,29 +65,29 @@ void vent_send_status(enum VALVE_STATE req_state) {
...
+        curr_state = VALVE_CLOSED;
+
+    // somewhere in between, but not close to either
+    } else if (LIN_ACT_LOWER_LIMIT_mV < pot_voltage_mV
+            && pot_voltage_mV < LIN_ACT_UPPER_LIMIT_mV) {
+        curr_state = VALVE_UNK;
+
+    // outside the voltage limits
+    } else {
+        curr_state = VALVE_ILLEGAL;
```
Comment (akmorrison) [inline lin_actuator.c:90]: I'll test this out on hardware tomorrow, but the logic seems correct to me

**Review by akmorrison — APPROVED** [review]
nothing major, it gets my approval. If you wanna add hysteresis to the critical warning that's your call, but it's not required.

File: lin_actuator.c (hunk 3 above)
Comment (amihaila) [reply lin_actuator.c]: Ah, I don't think I pulled today.

Comment (amihaila) [reply lin_actuator.c]: I didn't expect you to commit to master on my repo lol

**Pull request merged**